ecommerce/src/main/kotlin/io/hhplus/ecommerce/inventory/application/handler/InventoryEventHandler.kt

```
@@ -65,32 +67,41 @@ class InventoryEventHandler(
             } catch (e: InventoryException.InventoryNotFound) {
                 // 재고 없음 → 재고 부족과 동일하게 처리 (가용 재고 0)
                 val productId = e.data["productId"] as Long
-                val item = payload.items.find { it.productId == productId }
+                val reservation = reservations.find { it.productId == productId }
                 logger.warn("[InventoryEventHandler] 재고 없음: orderId=${payload.orderId}, productId=$productId")
-                publishInventoryInsufficientEvent(payload, productId, item?.quantity ?: 0, 0)
+                publishInventoryInsufficientEvent(payload, productId, reservation?.quantity ?: 0, 0)
             }
 
             true
         } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)